Pearson-Advance/edx-platform
# PADV-1418 feat: adds a new custom content jsonfield to ccx model

## Description

This PR adds a new custom content ``JSONField`` to the CCX ``CustomCourseForEdX`` model, in order to use this field to storage any additional information that we need to associate with.

## Testing instructions

1. Using your local environment with this branch, go to the LMS container and run the commands.
```
python manage.py lms makemigrations
python manage.py lms migrate
```
2. Go to **http://{base_url}/api/ccx/v0/ccx/?master_course_id={master_course_id}** to see the list of CCX courses associate to your master course. Look for the ``other_course_settings`` field for the CCX in the list. 
```
{
   "next":null,
   "previous":null,
   "count":7,
   "num_pages":1,
   "current_page":1,
   "start":0,
   "results":[
   {
      "ccx_course_id":"ccx-v1:demo+cs100+2024-t2+ccx@1",
      "master_course_id":"course-v1:demo+cs100+2024-t2",
      "display_name":"skillable edunext test ccx",
      "coach_email":"edx@example.com",
      "start":"2024-09-24 13:17:07+00:00",
      "due":null,
      "max_students_allowed":200,
      "course_modules":[],
      "other_course_settings":{}
   }, ...]}
```
3. Go to **http://{base_url}/api/ccx/v0/ccx/{ccx_course_id}** to see the details for a specific CCX course. Again, look for the ``other_course_settings`` field
4. Using Postman, you can test PUT and PATCH request to add/modify the ``other_course_settings`` field.

Note: Here is a Postman collection to import and test easily 
[Custom content XL-local.postman_collection.json](https://github.com/user-attachments/files/17277438/Custom.content.XL-local.postman_collection.json)


## Jira issue

* https://agile-jira.pearson.com/browse/PADV-1418
